io/eolearn/tests/test_processing_io.py

```diff
@@ -229,6 +229,24 @@ def test_additional_data(self):
         self.assertTrue(sun_zenith_angles.shape == (4, height, width, 1))
         self.assertTrue(len(eopatch.timestamp) == 4)
 
+    def test_aux_input_args(self):
+        """ Download low resolution data with `PREVIEW` mode
+        """
+        task = SentinelHubInputTask(
+            bands_feature=(FeatureType.DATA, 'BANDS'),
+            resolution=260,
+            maxcc=self.maxcc,
+            time_difference=self.time_difference,
+            data_source=DataSource.SENTINEL2_L1C,
+            max_threads=self.max_threads,
+            aux_input_args={'dataFilter': {'previewMode': 'PREVIEW'}}
+        )
+
+        eopatch = task.execute(bbox=self.bbox, time_interval=self.time_interval)
+        bands = eopatch[(FeatureType.DATA, 'BANDS')]
+
+        self.assertTrue(bands.shape == (4, 4, 4, 13))
+
```

Comment:
Technically, this test has little to no effect whatsoever. I know this because changes in sentinelhub-py have not been released yet. Therefore CI is using the previous sentinelhub-py version, the preview parameter gets ignored but the test still passes.

Could we add a check that would make this test fail if the preview parameter is ignored?

Reply:
> Could we add a check that would make this test fail if the preview parameter is ignored?

But this would then always fail and I don't see the benefit of checking for a raised exception which is there by design.

> Therefore CI is using the previous sentinelhub-py version, the preview parameter gets ignored but the test still passes.

I don't understand how this enables the test to pass. Can you elaborate? In any case, when the changes are released, this test will make sense, right?

Reply:
It passes because it doesn't 

> > Could we add a check that would make this test fail if the preview parameter is ignored?
> 
> But this would then always fail and I don't see the benefit of checking for a raised exception which is there by design.

I meant to add a check that would fail when running with the old sentinelhub-py version that ignores preview but will pass with the new version that actually includes the preview parameter in a request.

> 
> > Therefore CI is using the previous sentinelhub-py version, the preview parameter gets ignored but the test still passes.
> 
> I don't understand how this enables the test to pass. Can you elaborate? In any case, when the changes are released, this test will make sense, right?

The test passes because it only checks that it got something back from the service but not what. So whatever parameters you would specify the test would still pass. So does it really make sense to have a test that always passes?

Reply:
Resolved in private.